x-pack/plugins/security_solution/public/detections/components/detection_page_filters/index.tsx

```diff
@@ -32,19 +43,30 @@ const FilterItemSetComponent = (props: FilterItemSetProps) => {
     // applicable to `alert` page as new alert mappings are added when first alert
     // is encountered
     (async () => {
-      const dataView = await dataViewService.get(dataViewId ?? '');
-      if (!dataView) return;
+      const localDataViewId = title;
+      let dataView;
+      try {
+        dataView = await dataViewService.get(localDataViewId ?? '');
+      } catch (error) {
+        // creates an adhoc dataview if it does not already exists just for alert index
+        dataView = await dataViewService.create({
+          id: DATA_VIEW_NAME,
+          name: DATA_VIEW_NAME,
+          title: localDataViewId,
+          allowNoIndex: true,
+        });
+      }
       for (const filter of DEFAULT_DETECTION_PAGE_FILTERS) {
         const fieldExists = dataView.getFieldByName(filter.fieldName);
         if (!fieldExists) {
-          dataViewService.clearInstanceCache(dataViewId ?? '');
+          dataViewService.clearInstanceCache(localDataViewId ?? '');
```

Comment:
Tell me if am wrong here, but localDataViewId is not associated to a data view id, so the get will never work because  it is just the name of alert index. I think you will always need to do that 
```ts
dataView = await dataViewService.create({
          id: DATA_VIEW_NAME,
          name: DATA_VIEW_NAME,
          title,
          allowNoIndex: true,
        });
```
and then you can do 
```ts
dataViewService.clearInstanceCache(dataView?.id ?? '');
```

Reply:
I do not understand what we trying to do here
```ts
for (const filter of DEFAULT_DETECTION_PAGE_FILTERS) {
        const fieldExists = dataView.getFieldByName(filter.fieldName);
        if (!fieldExists) {
          // Remove clearing the cache here
          setLoadingPageFilters(false);
          return;
        }
      }
      setLoadingPageFilters(false);
```

Reply:
Thanks for the review @XavierM.

So what I observed previously with dataViewService is that it caches the result and for us it is necessary sometimes that the result is not stale. Consider below scenario for example

1. User logs into a new space for the first time in alerts
2. We create a dataview with the `alerts` index.
3. But currently mappings do not exists so the Alert Page Controls ( dropdown ) will be error state since the fields will not be available.
4. User creates a rule, and when the first alerts are found, index is updated with the mapping and the alerts.
5. Now, assumption is dataviewService has already cached the previous result ( `with index without mapping` ), that is why we check that if we field do not exist, we clear the cache which is the purpose of below code:

```javascript
for (const filter of DEFAULT_DETECTION_PAGE_FILTERS) {
        const fieldExists = dataView.getFieldByName(filter.fieldName);
        if (!fieldExists) {
          // Remove clearing the cache here
          dataViewService.clearInstanceCache(localDataViewId ?? '');
          setLoadingPageFilters(false);
          return;
        }
      }
      setLoadingPageFilters(false);
```
7. And we also check that in case the temporary dataview already exists, we do not create that.

To answer your questions:

> Tell me if am wrong here, but localDataViewId is not associated to a data view id, so the dataViewService.get will never work because it is just the name of alert index. I think you will always need to do that

You are correct it should be ` dataView = await dataViewService.get(DATA_VIEW_NAME)`. It is being set as ID in below 
```js
        dataView = await dataViewService.create({
          id: DATA_VIEW_NAME,
          name: DATA_VIEW_NAME,
          title: localDataViewId,
          allowNoIndex: true,
        });
```

Please let me know if any of the assumption is incorrect here.

Reply:
@XavierM , could you please check final changes. Thanks.